File: my-folder/problems/3sum/solution.py
class Solution(object):
    def threeSum(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        nums=sorted(nums)
        def findsum(num,i,ot):
            j=len(num)-1
            k=i+1
            while k<j:
                if num[k]+num[j]+num[i]==0:
                    if [num[i],num[k],num[j]] not in ot:
                        ot.append([num[i],num[k],num[j]])
                    k+=1
                if num[k]+num[j]+num[i]>0:
                    j-=1
                if num[k]+num[j]+num[i]<0:
                    k+=1
        ot=[]
        for i in range(len(nums)):
            if nums[i] > 0:
                break
            if i==0 or nums[i-1]!=nums[i]:
                findsum(nums,i,ot)
        return ot
        

File: my-folder/problems/3sum/solution.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        Set<List<Integer>> resp = new HashSet();
        for (int i = 0; i < nums.length; i++) {
            if (i > 0 && nums[i] == nums[i - 1]) continue;
            if (nums[i]>0) {
                break;
            }
            int[] temparray = Arrays.copyOfRange(nums, i+1, nums.length);
            List<List<Integer>> out = twoSum(temparray, -1*nums[i]);
            if(out.size() !=0){
                for (int j = 0; j < out.size(); j++) {
                    resp.add(new ArrayList(Arrays.asList(nums[i],nums[i+1+out.get(j).get(0)],nums[i+1+out.get(j).get(1)])));
                }
            }
        }
        return new ArrayList(resp);
    }

    public List<List<Integer>> twoSum(int[] nums, int target) {
        Set<List<Integer>> resp = new HashSet<>();
        Map<Integer,Integer> mp=new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (mp.containsKey(target-nums[i])){
                resp.add(Arrays.asList(mp.get(target-nums[i]),i));
            }
            mp.put(nums[i], i);
        }
        return new ArrayList(resp);
    }
}